hermes/probe/create_file.go

```diff
@@ -0,0 +1,116 @@
+package probe
+<<<<<<< HEAD
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/google/cloudprober/targets/endpoint"
+	"github.com/googleinterns/step224-2020/blob/evan/hermesMain/cmd"
+)
+
+type RandomFile struct {
+	Seed int64
+	Size int // File size in bytes
+}
+
+func (f *RandomFile) NewReader() *randomFileReader {
+	return &randomFileReader{size: f.Size, rand: rand.New(rand.NewSource(f.Seed))}
```

Comment:
You're going to want to use a lockedSource to make this thread-safe. Sources aren't thread-safe by default.

```
type lockedSource struct {
	lk  sync.Mutex
	src *rngSource
}
```

See: https://golang.org/src/math/rand/rand.go

Reply:
It's unlikely that a Reader instances would be used in multiple threads.
Also note that only Hermes is going to use this: just add a comment stating that it's not thread safe.

Reply:
If the reader itself has a rand variable with a source, can multiple readers exist with different sources with conflicting? If so, then there's no issue with this.

Hermes will be using an instance of this Reader in each probe iteration, so there would be multiple in existence if multiple targets are being monitored.

Reply:
NewReader() returns a new instance every time. Each instance is independent.
Each instance is going to be used for a short period for a single purpose: either to calculate the checksum or to send the contents to the storage system.
I don't expect these instances to be retained, so no locking is necessary.